components/telemetry-operator/controllers/logpipeline_controller.go

```diff
@@ -150,7 +149,7 @@ func (r *LoggingConfigurationReconciler) syncConfigMap(ctx context.Context, conf
 			controllerutil.RemoveFinalizer(config, configMapFinalizer)
 		}
 	} else {
-		fluentBitConfig := generateFluentBitConfig(config.Spec.Sections)
+		fluentBitConfig := mergeFluentBitConfig(config.Spec.Parsers, config.Spec.MultiLineParsers, config.Spec.Filters, config.Spec.Outputs)
```

Comment:
IMHO, you could just pass the whole `LogPipeline`, no need to split it into 4 args.